tests/test_downloadermiddleware_stats.py

```diff
@@ -37,6 +39,17 @@ def test_process_response(self):
         self.mw.process_response(self.req, self.res, self.spider)
         self.assertStatsEqual('downloader/response_count', 1)
 
+    def test_response_len(self):
+        body = (b'', b'not_empty')  # empty/notempty body
+        headers = ({}, {'lang': 'en'}, {'lang': 'en', 'User-Agent': 'scrapy'})  # 0 headers, 1h and 2h
+        test_responses = [
+            Response(url='scrapytest.org', status=200, body=r[0], headers=r[1])
+            for r in product(body, headers)]  # form test responses with all combinations of body/headers
```

Comment:
:lipstick: 
```suggestion
        test_responses = [
            Response(
                url='https://scrapy.org/',
                status=200,
                body=body,
                headers=headers,
            )
            for body, headers in product(body, headers)
        ]
```

Reply:
@Gallaecio  fixed https://github.com/scrapy/scrapy/pull/4972/commits/fbdc6a9a25b3b457d509580349e286e5c5d5771e

Reply:
The used URL is still invalid (no scheme), and points to a domain we do not own which, based on recent experience, could eventually be a NSFW website.

Reply:
@Gallaecio I think that in this specific case - it is not an issue. This test creates `Response` objects from scratch without sending requests.